Month3/Project1/Project1/TDinamicArray.h

```diff
@@ -0,0 +1,290 @@
+		{
+			NewArray[CurrentIndex] = Array[CurrentIndex];
+		}
+		delete Array;
+
+		Capacity = Size;
+		Array = NewArray;
+	}
+
+	void Clear()
+	{
+		if (Size > 0)
+		{
+			Size = 0;
+		}
+	}
+
+
+	void PushBack(const T& NewShape)
+	{
+		if (Capacity == 0)
+		{
+			Reserve(1);
+			Size++;
+			Array[Size] = NewShape;
+		}
+		if (Size == Capacity)
+		{
+			int NewCapacity = Capacity * 2;
+			T* NewArray = new T[NewCapacity];
+			for (int CurrentIndex = 0; CurrentIndex <= Capacity; CurrentIndex++)
+			{
+				NewArray[CurrentIndex] = Array[CurrentIndex];
+			}
+			Capacity = NewCapacity;
+			delete Array;
+			Array = NewArray;
+			Size += 1;
+			Array[Size] = NewShape;
+		}
+	}
+
+
+	void PopBack()
+	{
+		Size -= 1;
+	}
+
+	void Insert(int Index, const T& Shape)
+	{
+		std::cout << "In which position do you want to add your Shape?" << std::endl;
+		std::cin >> Index;
+		if (Capacity == Index)
+		{
+			int NewCapacity = Capacity * 2;
+			T* NewArray = new T[NewArray];
+			for (int CurrentIndex = 0; CurrentIndex <= Capacity; CurrentIndex++)
+			{
+				NewArray[CurrentIndex] = Array[CurrentIndex];
+			}
+			delete Array;
+
+			Capacity = NewCapacity;
+			Array = NewArray;
+			if (Size == Index)
+			{
+				Array[Size + 1] = Array[Size];
+				Array[Index] = Shape;
+			}
+			Size += 1;
+
+		}
+		if (Capacity == Size)
+		{
+			int NewCapacity = Capacity * 2;
+			T* NewArray = new T[NewArray];
+			for (int CurrentIndex = 0; CurrentIndex <= Capacity; CurrentIndex++)
+			{
+				NewArray[CurrentIndex] = Array[CurrentIndex];
+			}
+			delete Array;
+
+			Capacity = NewCapacity;
+			Array = NewArray;
+			for (int CurrentIndex = Size; CurrentIndex >= Index; CurrentIndex--)
+			{
+				Array[CurrentIndex + 1] = Array[CurrentIndex];
+			}
+			Size++;
+			Array[Index] = Shape;
+		}
+
+	}
+
+	void Erase(int Index)
+	{
+		std::cout << "In which position do you want to delete the Shape?" << std::endl;
+		std::cin >> Index;
+		if (Index < Size && Size > 1)
+		{
+			for (int CurrentIndex = Index; CurrentIndex < Size - 1; CurrentIndex++)
+			{
+				Array[CurrentIndex] = Array[CurrentIndex + 1];
+
+			}
+		}
+		if (Size == Capacity && Size == 1)
+		{
+			Size = 0;
+		}
+		if (Index == Size)
+		{
+			Size -= 1;
+		}
+	}
+
+
+	void Resize(int NewCapacity)
+	{
+		std::cout << "What size do you want your arrangement to be now?" << std::endl;
+		std::cin >> NewCapacity;
+		T* NewArray = new T[NewCapacity];
+
+		for (int CurrentIndex = 0; CurrentIndex <= NewCapacity;CurrentIndex++)
+		{
+			NewArray[CurrentIndex] = Array[CurrentIndex];
+		}
+		delete Array;
+		Capacity = NewCapacity;
+		Array = NewArray;
+	}
+
+	void Append(const TDinamicArray<T>& Array1) 
+	{
+		Reserve(Array1.Size);
+		for (int I = 0; I < Array1.Size; I++) 
+		{
+			PushBack(Array1[I]);
+		}
+	}
+
+	void FillRandSquare(T& Filler)
+	{
+		srand(time(0));
+		float NewSideLenght;
+		NewSideLenght = rand();
+		T* Array = new FSquare(NewSideLenght);
+	}
+	bool ItIsFull() 
```

Comment:
I think a better name for this might be `IsFull()`. A little shorter and still has the same meaning.